src/items/associated-items.md

```diff
@@ -0,0 +1,260 @@
+# Associated Items
+
+*Associated Items* are the items declared in [traits] or defined in
+[implementations]. They are called this because they are defined on an associate
+type &mdash; the type in the implementation. They are a subset of the kinds of
+items you can declare in a module. Specifically, there are [associated
+functions], [associated types], and [associated constants].
+
+[associated functions]: #associated-functions-and-methods
+[associated types]: #associated-types
+[associated constants]: #associated-constants
+
+Associated items are useful when the associated item logically is related to the
+associating item. For example, the `is_some` method on `Option` is intrinsically
+related to Options, so should be associated.
+
+Associated items are also the contract that traits have. 
+
+## Associated functions and methods
+
+*Associated functions* are [functions] associated with a type.
+
+An *associated function declaration* is written as `fn`, then an [identifier]
```

Comment:
This applies throughout the doc: spelling grammar out explicitly in words seems verbose and hard to read. Moreover, this seems redundant with regular function grammar. Do we have a BNF standard we can use? Or perhaps update the grammar page and link to that?